src/main/java/cardibuddy/model/Model.java

```diff
@@ -78,11 +78,44 @@
      * {@code target} must exist in the address book.
      * The person identity of {@code editedFlashcard} must not be the same as another existing person in the address book.
      */
-    void setDeck(Flashcard target, Flashcard editedFlashcard);
+    void setDeck(Deck target, Deck editedDeck);
+
+    /**
+     * Returns true if a person with the same identity as {@code person} exists in the address book.
+     */
+    boolean hasCard(Flashcard card);
```

Comment:
Sorry about my naming last time. I can change the flashcard methods to hasFlashcard etc. to match the Cardibuddy class if you want.

Reply:
up to you!!